homeassistant/components/volvooncall.py

```diff
@@ -148,29 +202,41 @@ def vehicle_name(self, vehicle):
 class VolvoEntity(Entity):
     """Base class for all VOC entities."""
 
-    def __init__(self, hass, vin, attribute):
+    def __init__(self, hass, vin, component, attribute):
         """Initialize the entity."""
-        self._hass = hass
-        self._vin = vin
-        self._attribute = attribute
-        self._state.entities[self._vin].append(self)
+        self.vin = vin
+        self.component = component
+        self.attribute = attribute
+        self._data = hass.data[DATA_KEY]
```

Comment:
Pass in the `VolvoData` data instance directly when creating the entity instead of passing in `hass`.

Reply:
✓